### PR TITLE
add a stub of etcdctlv3 auth enable

### PR DESCRIPTION
**Pull request by mitake**
This PR  adds a stub of etcdctlv3 auth enable. Related RPC stubs are also implemented.

### CODE REVIEW

File: etcdserver/v3demo_server.go [hunk 1]
```diff
@@ -57,6 +57,10 @@ type Lessor interface {
 	LeaseRenew(id lease.LeaseID) (int64, error)
 }
 
+type AuthManager interface {
```
Comment (xiang90) [inline etcdserver/v3demo_server.go]: authenticator?

**Comment by xiang90** [comment]
remove the binary etcdctlv3? (probably add that into gitignore?)

File: etcdserver/v3demo_server.go (hunk 1 above)
Comment (mitake) [inline etcdserver/v3demo_server.go]: Authenticator seems to be a good name. I renamed it.

**Comment by mitake** [comment]
@xiang90 sorry, removed the binary and added the gitignore file

File: etcdctlv3/command/auth_command.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	"fmt"
+
+	"github.com/coreos/etcd/Godeps/_workspace/src/github.com/spf13/cobra"
+	"github.com/coreos/etcd/Godeps/_workspace/src/golang.org/x/net/context"
+)
+
+// NewAuthCommand returns the cobra command for "auth".
+func NewAuthCommand() *cobra.Command {
+	ac := &cobra.Command{
+		Use:   "auth",
```
Comment (xiang90) [inline etcdctlv3/command/auth_command.go]: auth <"enable" or "disable">

Comment (xiang90) [inline etcdctlv3/command/auth_command.go]: then we do not need the sub command?

Comment (xiang90) [inline etcdctlv3/command/auth_command.go]: or probably we should do `auth enable true/false`, `auth user ...`, `auth role ...`? Then we should change the short description of this command.

Comment (mitake) [inline etcdctlv3/command/auth_command.go]: I think following the CLI of v2 would be good for users. How about changing the `Use` field to `auth <"enable" or "disable"> and keeping the subcommand?

**Comment by mitake** [comment]
@xiang90 updated the description of `auth` command

File: etcdctlv3/command/auth_command.go (hunk 2 above)
Comment (xiang90) [inline etcdctlv3/command/auth_command.go]: we do not have to follow v2. i am not sure why do we need the subcommand if we can do one command with a required argument.

/cc @heyitsanthony opinion?

**Comment by xiang90** [comment]
LGTM

**Comment by xiang90** [comment]
defer to @heyitsanthony

**Comment by heyitsanthony** [comment]
lgtm

**Pull request merged**